tests/test_defects.py

```diff
@@ -103,30 +108,472 @@ def setUp(self):
+                     Box2I(corner = Point2I(50, 40), dimensions = Extent2I(1, 1))]
+        for badBox in badPixels:
+            defects.append(badBox)
+
+        self.check_maskBlocks(defects, expectedDefects)
+
+    def test_maskBlocks_every_other_pixel_bad_less_than_threshold(self):
+        """A test for maskBlocksIfIntermitentBadPixelsInColumn (test 9).
+        Npix discontiguous bad pixels in a column where Npix > badOnAndOffPixelColumnThreshold (10)
+        and every other pixel is bad.
+        Input defects for this test: [55, 20, 1, 1], [55, 22, 1, 1], [55, 24, 1, 1], [55, 26, 1, 1],
+                                     [55, 28, 1, 1], [55, 30, 1, 1]
+
+        Expected defects after the function runs: [55, 20, 1, 1], [55, 22, 1, 1], [55, 24, 1, 1],
+                                                  [55, 26, 1, 1], [55, 28, 1, 1], [55, 30, 1, 1]
+
+        Plots can be found in DM-19903 on Jira.
+        """
+
+        expectedDefects = [Box2I(corner = Point2I(55, 20), dimensions = Extent2I(1, 1)),
+                           Box2I(corner = Point2I(55, 22), dimensions = Extent2I(1, 1)),
+                           Box2I(corner = Point2I(55, 24), dimensions = Extent2I(1, 1)),
+                           Box2I(corner = Point2I(55, 26), dimensions = Extent2I(1, 1)),
+                           Box2I(corner = Point2I(55, 28), dimensions = Extent2I(1, 1)),
+                           Box2I(corner = Point2I(55, 30), dimensions = Extent2I(1, 1))]
+        defects = self.allDefectsList
+        badPixels = [Box2I(corner = Point2I(55, 20), dimensions = Extent2I(1, 1)),
+                     Box2I(corner = Point2I(55, 22), dimensions = Extent2I(1, 1)),
+                     Box2I(corner = Point2I(55, 24), dimensions = Extent2I(1, 1)),
+                     Box2I(corner = Point2I(55, 26), dimensions = Extent2I(1, 1)),
+                     Box2I(corner = Point2I(55, 28), dimensions = Extent2I(1, 1)),
+                     Box2I(corner = Point2I(55, 30), dimensions = Extent2I(1, 1))]
+        for badBox in badPixels:
+            defects.append(badBox)
+
+        self.check_maskBlocks(defects, expectedDefects)
+
+    def test_maskBlocks_blobs_one_side_good_less_than_threshold(self):
+        """A test for maskBlocksIfIntermitentBadPixelsInColumn (test 10).
+        Npix discontiguous bad pixels in column with "blobs" of "m" bad pixels to one side,
+        m > badOnAndOffPixelColumnThreshold (10), number of good pixel in gaps between blobs <
+        goodPixelColumnGapThreshold (5).
+
+        Input defects for this test: (60, 1, 1, 18), (60, 20, 1, 10), (61, 2, 2, 2), (61, 6, 2, 8)
+
+        Expected defects after the function runs: (60,1,1,31), (61, 2, 2, 14)
+
+        Plots can be found in DM-19903 on Jira.
+        """
+
+        expectedDefects = [Box2I(corner = Point2I(60, 1), dimensions = Extent2I(1, 29)),
+                           Box2I(corner = Point2I(61, 2), dimensions = Extent2I(1, 12))]
```

Comment:
I don't think this is right: the extent of that region is 2 wide in 'x', so the output bad block should have `Extent(2,12)`, I believe. If this is what the code was producing, there may be a bug.

Reply:
Thanks for catching this. I did run the tests and they all passed before asking you to look at the code, so the code per se it's not incorrect, but rather incomplete in the sense that there was a column missing from the "expectedDefects" set. For the case of this test, I have added this extra expected defect (Box2I(corner = Point2I(62, 2), dimensions = Extent2I(1, 12))),  but I also added a few more for tests 11, 12, and 15, for the same reason.